### PR TITLE
Use jolicode/castor=^0.17.0

### PR DESCRIPTION
**Pull request by maximehuran**
- [x] Fix Build macOS pipeline

### CODE REVIEW

File: .github/actions/install/action.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        description: 'The PHP version to use'
+        required: true
+runs:
+    using: 'composite'
+    steps:
+        -   name: Setup PHP
+            uses: shivammathur/setup-php@v2
+            with:
+                php-version: ${{ inputs.php }}
+                tools: box
```
Comment (jacquesbh) [inline .github/actions/install/action.yaml:14]:
```suggestion
                tools: box castor
```

File: .github/workflows/pipeline.yaml [hunk 2]
```diff
@@ -16,11 +16,14 @@ jobs:
...
             -   uses: actions/checkout@v4
+            -   uses: ./.github/actions/install
+                with:
+                    php: ${{ env.PHP_VERSION }}
             -   name: Castor cache
                 uses: actions/cache@v4
                 id: cache-castor-macos
                 with:
-                    path: /Users/runner/.cache/castor
+                    path: /home/runner/.cache/castor
```
Comment (jacquesbh) [inline .github/workflows/pipeline.yaml:26]: @maximehuran This is why your macOS pipeline fails.

On macOS the home path is `/Users/`, not `/home/`.

**Comment by maximehuran** [comment]
@jacquesbh I have an issue with static PHP if you know why https://github.com/monsieurbiz/SyliusPluginMaker/actions/runs/9527809570/job/26264831724?pr=5

**Comment by jacquesbh** [comment]
@maximehuran This is the same issue as the one in castor's pipeline.

**Pull request merged**